src/Interpreters/Cache/FileCacheSettings.cpp

```diff
@@ -32,6 +33,20 @@ void FileCacheSettings::loadImpl(FuncHas has, FuncGetUInt get_uint, FuncGetStrin
     if (max_size == 0)
         throw Exception(ErrorCodes::BAD_ARGUMENTS, "Expected non-zero size for cache configuration");
 
+    try
+    {
+        auto fs_info = std::filesystem::space(base_path);
```

Comment:
this path could not exist yet, it would be created if it does not exist here https://github.com/ClickHouse/ClickHouse/blob/602ecc3b8dada6642c270eff122edf5d8e9fe826/src/Interpreters/Cache/FileCache.cpp#L186-L190
so because of this there are now a few failing checks in ci.
So I'd suggest to move this check there.